hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -3864,23 +3886,130 @@ private void getProcedureResult(long procId, CompletableFuture<Void> future, int
+  public CompletableFuture<Boolean> balanceRSGroup(String groupName) {
+    return this.<Boolean> newMasterCaller()
+        .action((controller, stub) -> this.
+            <BalanceRSGroupRequest, BalanceRSGroupResponse, Boolean> call(controller, stub,
+                BalanceRSGroupRequest.newBuilder().setRSGroupName(groupName).build(),
+              (s, c, req, done) -> s.balanceRSGroup(c, req, done),
+                resp -> resp.getBalanceRan()))
+        .call();
+  }
+
+  @Override
+  public CompletableFuture<List<RSGroupInfo>> listRSGroups() {
+    return this.<List<RSGroupInfo>> newMasterCaller()
         .action((controller, stub) -> this
-            .call(controller, stub,
-                RequestConverter.buildIsSnapshotCleanupEnabledRequest(),
-                MasterService.Interface::isSnapshotCleanupEnabled,
-                IsSnapshotCleanupEnabledResponse::getEnabled))
+            .<ListRSGroupInfosRequest, ListRSGroupInfosResponse, List<RSGroupInfo>> call(controller,
+                stub, ListRSGroupInfosRequest.getDefaultInstance(),
+                (s, c, req, done) -> s.listRSGroupInfos(c, req, done),
+                resp -> resp.getRSGroupInfoList().stream()
+                    .map(r -> ProtobufUtil.toGroupInfo(r))
+                    .collect(Collectors.toList())))
+        .call();
+  }
+
+  @Override
+  public CompletableFuture<RSGroupInfo> getRSGroupOfServer(Address hostPort) {
+    return this.<RSGroupInfo> newMasterCaller()
+        .action((controller, stub) -> this.
+            <GetRSGroupInfoOfServerRequest, GetRSGroupInfoOfServerResponse, RSGroupInfo> call(
+                controller, stub,
+               RequestConverter.buildGetRSGroupInfoOfServerRequest(hostPort),
+              (s, c, req, done) -> s.getRSGroupInfoOfServer(c, req, done),
+              resp -> resp.hasRSGroupInfo() ?
+                  ProtobufUtil.toGroupInfo(resp.getRSGroupInfo()) : null))
+        .call();
+  }
+
+  @Override
+  public CompletableFuture<Void> removeServers(Set<Address> servers) {
+    return this.<Void> newMasterCaller()
+        .action((controller, stub) -> this.
+            <RemoveServersRequest, RemoveServersResponse, Void> call(controller, stub,
+                RequestConverter.buildRemoveServersRequest(servers),
+              (s, c, req, done) -> s.removeServers(c, req, done), resp -> null))
+        .call();
+  }
+
+  @Override
+  public CompletableFuture<Void> setRSGroupForTables(Set<TableName> tables, String groupName) {
+    return this.<Void> newMasterCaller()
+        .action((controller, stub) -> this.
+            <SetRSGroupForTablesRequest, SetRSGroupForTablesResponse, Void> call(controller, stub,
+                RequestConverter.buildSetRSGroupForTablesRequest(tables, groupName),
+              (s, c, req, done) -> s.setRSGroupForTables(c, req, done), resp -> null))
+        .call();
+  }
+
+  @Override
+  public CompletableFuture<RSGroupInfo> getRSGroupInfoOfTable(TableName table) {
```

Comment:
I think this can be done at client side? Just get the table descriptor and get its region server group config.

Reply:
Yes, I'll do that.